application/client.core/src/app/environment/components/complex/popups/directive/popups.directive.ts

```diff
@@ -0,0 +1,14 @@
+import { Directive, ElementRef, OnInit } from '@angular/core';
+
+@Directive({
+    selector: '[appComplexPopups]',
+})
+
+export class ComplexPopupsDirective implements OnInit {
+
+    constructor(private element: ElementRef) { }
+
+    ngOnInit() {
+        this.element.nativeElement.focus();
+    }
```

Comment:
Event focus can be applied to such kind of HTMLElements like Input, TextArea, Select, CheckBox and other forms controls. But it also possible to "focus" any other node (like `div`, or `span`). To do it right node should have attribute `tabindex` (https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex)

Like
```
<div tabindex="0">Tabbable due to tabindex.</div>
```

Reply:
One more thing. Please check: as far as I remember `OnInit` called after all data is bound, but it doesn't give a guaranty view is ready. Probably would be safer to use `AfterViewInit` here? Because this hook would be called only after the whole view with children is checked and ready